static/js/components/forms/ChangePasswordForm.js

```diff
@@ -39,20 +57,31 @@ const ChangePasswordForm = ({ onSubmit, user }: Props) => (
             <Field
               name="email"
               className="form-control"
-              readOnly={true}
               component={EmailInput}
             />
-            <ErrorMessage name="oldPassword" component={FormError} />
+            <ErrorMessage name="email" component={FormError} />
+          </div>
+          <div className="form-group hidden">
+            <label htmlFor="user_email">User Email</label>
+            <Field
+              name="user_email"
+              className="form-control"
+              component={EmailInput}
+              value={user.email}
+            />
+            <ErrorMessage name="user_email" component={FormError} />
+          </div>
```

Comment:
Why is there a whole hidden input here? I think it's being using for validation - if that's the case we should use `yup`'s context functionality w/ `when()` to accomplish that instead.

Reply:
It's UX issue and I'm using it when for that case when you email is same in the form, user going to change password and wise versa.

That's hidden field value is being used in the validation with `when` for email comparison.

Reply:
`emailPassword: yup
    .string()
    .label("Confirm Password")
    .when(["email", "user_email"], (email, user_email, schema) =>
      user_email !== email ? schema.required().min(8) : schema.notRequired()
    ),`

Reply:
ok, then it should use the context functionality I mentioned, then you just pass the current email when you call `validate()`. See the docs for an example: https://github.com/jquense/yup#mixedwhenkeys-string--arraystring-builder-object--value-schema-schema-schema

Reply:
I had tried that twice but was not working out. So I went for keeping the hidden field for the user email field for later use of comparison. But I will try it one more time on your suggestion. I might be doing it wrong.

Reply:
It doesn't work out, it only picks siblings fields that's why I had added that field.

Reply:
Did you try this? it works for me: https://gist.github.com/rhysyngsun/a249fba0f0577c684f00c6a0a9513d88

Reply:
It's not working on my MacBook, `currentEmail` giving `null`, let me try it on one of the folk's system.

Reply:
See its asking for `emailPassword` even if I have not changed the email.  It should go for checking validation only if I have changed the email address.

<img width="1437" alt="Screenshot 2020-02-25 at 19 52 58" src="https://user-images.githubusercontent.com/4043989/75258807-d6814480-5808-11ea-8a3f-04681fe62130.png">

Reply:
Ok, I dug into this because something was off since it was passing the tests for `lib/validation.js`.  The issue is this form is getting validated _twice_. Once in the function passed into `<Formik validate={}/>` and again by formik itself using the schema passed into `<Formik validationSchema={}/>`. If you apply my changes and then remove `validationSchema` then you won't get that error anymore.

Reply:
Actually it ended up involving more than just that. `validate()` need to mimic the internal behavior of formik validation. Since we need to set a `context`, we still need to use `validate`. Formik has two helper functions it exports that run the validation and transform the error into an errors object, so I've updated the gist with what I found works locally for me. I think this should work on your end too: 

https://gist.github.com/rhysyngsun/a249fba0f0577c684f00c6a0a9513d88

Reply:
that seems to be working fine on my system too.